nodeadm/internal/configprovider/userdata.go

```diff
@@ -83,23 +88,27 @@ func parseMultipart(userDataReader *multipart.Reader) (*internalapi.NodeConfig,
 		if err != nil {
 			return nil, err
 		}
-		if partHeader := part.Header.Get("Content-Type"); partHeader != "" {
+		if partHeader := part.Header.Get(contentTypeHeader); len(partHeader) > 0 {
 			mediaType, _, err := mime.ParseMediaType(partHeader)
 			if err != nil {
 				return nil, err
 			}
 			if mediaType == nodeConfigMediaType {
-				if config != nil {
-					return nil, fmt.Errorf("Found multiple node bootstrap configurations in the UserData")
-				}
 				nodeConfigPart, err := io.ReadAll(part)
 				if err != nil {
 					return nil, err
 				}
-				config, err = apibridge.DecodeNodeConfig(nodeConfigPart)
+				decodedConfig, err := apibridge.DecodeNodeConfig(nodeConfigPart)
 				if err != nil {
 					return nil, err
 				}
+				if config != nil {
+					if err := config.Merge(decodedConfig); err != nil {
+						return nil, err
+					}
+				} else {
+					config = decodedConfig
+				}
```

Comment:
I think it'd be a bit clearer to just collect the decoded configs in a slice and then merge them all at the end.

Reply:
done!